### ISSUE TITLE
Environment variable interpolation is not supported

### ISSUE THREAD

**Issue by lpakula** [kedacore/keda#717]
Environment variable is not properly recognized when using interpolation from another environment variable
```
# Deployment
...
env: 
 - name: BROKER_URL
     valueFrom:
       secretKeyRef:
         name: rabbitmq
         key: BROKER_URL
 - name: BROKER_URL_VHOST
   value: "$(BROKER_URL)<vhost>"

# ScaledObject
...
  triggers:
  - type: rabbitmq
    metadata:
      queueName: celery
      host: BROKER_URL_VHOST
      queueLength  : '20'
```

## Expected Behavior
Because this approach is supported natively by kubernetes, it should also work with keda.

Environment variable in kubernetes container is correct
```
echo $BROKER_URL_VHOST
amqp://<user>:<password>@<host>:<port>/<vhost>
```

## Actual Behavior

Keda is interpreting the value as a literal string `$(...` and it's failing with the following error:

`"error getting scaler for trigger #0: error establishing rabbitmq connection: AMQP scheme must be either 'amqp://' or 'amqps://'`

This is a common use case to keep broker main creds in secret and apply vhost as a separate variable to avoid duplication.

Documentation says:

> The host value is the name of the environment variable your deployment uses to get the connection string.

But it's not working as expected

## Specifications

  - **KEDA Version:** *CHART: keda-1.3.1 APP VERSION: 1.3.0*
  - **Scaler(s):** *RabbitMQ*

**Comment by tomkerkhove** [comment]
We are actually looking at providing full connection, or splitting the parts into smaller options to configure (optional) but in your scenario `connection` & `vhost` are actually what you need, no?

**Comment by lpakula** [comment]
Yes it could be resolved by allowing separate `connection` and `vhost` configuration options, however, the documentation states that any environment is supported and it's not the case

**Comment by tomkerkhove** [comment]
Re-adding bug since this should be supported as it works natively in Kubernets. THanks for updating the issue

**Comment by jeffhollan** [comment]
@lpakula does it work when you don't use this syntax in the env `$(BROKER_URL)<vhost>` but instead just try adding the literal value (e.g. `amqp://<user>:<password>@<host>:<port>/<vhost>`).  Trying to see if it's an issue with scaler not supporting this format or with KEDA resolving the env value

**Comment by coderanger** [comment]
I'm still on 1.x but if I'm reading this right, #1236 should have fixed this in 2.0?

**Comment by zroubalik** [comment]
@coderanger yes!
